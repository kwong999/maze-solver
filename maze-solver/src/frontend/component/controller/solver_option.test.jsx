import React from 'react';
import renderer from 'react-test-renderer';
import SolverOption from './solver_option';

describe('SolverOption', () => {
  const component = renderer.create(
    <SolverOption
      movement={'All Direction'}
      solver={'A* Star'}
      changeSolver={jest.fn()}
      solverFull={jest.fn()}
      fullReset={jest.fn()}
      softReset={jest.fn()}
    />
  )
  test('should render without error', () => {
    let tree = component.toJSON();
    let componentInstance = component.root;
    expect(componentInstance.findByType('select').props.value).toBe('A* Star');
    let optionList = componentInstance.findAllByType('option');
    expect(optionList[0].props.value).toBe('A* Star');
    expect(optionList[1].props.value).toBe('Breadth First Search');
    expect(tree).toMatchSnapshot();
  })
})