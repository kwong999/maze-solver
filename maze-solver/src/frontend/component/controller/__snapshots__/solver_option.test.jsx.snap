// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SolverOption should render without error 1`] = `
<div
  className="maze-action"
>
  <div>
    <p>
      Solver: 
    </p>
    <select
      onChange={[MockFunction]}
      value="A* Star"
    >
      <option
        value="A* Star"
      >
        A* Star
      </option>
      <option
        value="Breadth First Search"
      >
        Breadth First Search
      </option>
    </select>
  </div>
  <button
    onClick={[Function]}
  >
    Solve It!
  </button>
  <button
    onClick={[MockFunction]}
  >
    Full Reset
  </button>
  <button
    onClick={[MockFunction]}
  >
    Soft Reset
  </button>
</div>
`;
